java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/UserdataService.java

```diff
@@ -211,8 +230,42 @@ public ResponseEntity<String> uploadMultipartFile(MultipartFile file, PremiumUse
                 throw new OsmAndPublicApiException(ERROR_CODE_GZIP_ONLY_SUPPORTED_UPLOAD, "File is submitted not in gzip format");
 			}
 		}
+        error = validateFreeAccountFile(pu, zipfile);
+        if (error != null) {
+            return error;
+        }
+        
 		return uploadFile(zipfile, dev, name, type, clienttime);
 	}
+    
+    private ResponseEntity<String> validateFreeAccount(PremiumUsersRepository.PremiumUser pu, String type) {
+        if (pu.orderid == null) {
```

Comment:
all account without valid subscription should be free -  userSubService.checkOrderIdPremium(pu.orderid); we should check 
```
 if (pu.orderid == null || !userSubService.checkOrderIdPremium(pu.orderid)) {
```